src/encoding/resolve.rs

```diff
@@ -0,0 +1,463 @@
+                        pos = stream_offset;
+                    }
+                    for glyph in &self.glyph_ranges[glyphs.clone()] {
+                        let glyph_data = &self.glyph_cache.encoding.linewidths
+                            [glyph.start.linewidths..glyph.end.linewidths];
+                        data.extend_from_slice(bytemuck::cast_slice(glyph_data));
+                    }
+                }
+            }
+            if pos < stream.len() {
+                data.extend_from_slice(bytemuck::cast_slice(&stream[pos..]));
+            }
+        }
+        layout.n_draw_objects = layout.n_paths;
+        assert_eq!(capacity, data.len());
+        (layout, self.ramp_cache.ramps())
+    }
+
+    fn resolve_patches(&mut self, encoding: &Encoding) -> StreamOffsets {
+        self.ramp_cache.advance();
+        self.glyph_cache.clear();
+        self.glyph_ranges.clear();
+        self.patches.clear();
+        let mut sizes = StreamOffsets::default();
+        for patch in &encoding.patches {
+            match patch {
+                Patch::Ramp { offset, stops } => {
+                    let ramp_id = self.ramp_cache.add(&encoding.color_stops[stops.clone()]);
+                    self.patches.push(ResolvedPatch::Ramp {
+                        draw_data_offset: *offset + sizes.draw_data,
+                        ramp_id,
+                    });
+                }
+                Patch::GlyphRun { index } => {
+                    let mut run_sizes = StreamOffsets::default();
+                    let run = &encoding.glyph_runs[*index];
+                    let font_id = run.font.data.id();
+                    let font_size_u32 = run.font_size.to_bits();
+                    let Some(font) = FontRef::from_index(run.font.data.as_ref(), run.font.index) else { continue };
+                    let glyphs = &encoding.glyphs[run.glyphs.clone()];
+                    let _coords = &encoding.normalized_coords[run.normalized_coords.clone()];
+                    let vars: [(moscato::pinot::types::Tag, f32); 0] = [];
+                    let hint_id = if run.font.index < 0xFF {
+                        Some(font_id << 8 | run.font.index as u64)
+                    } else {
+                        None
+                    };
+                    let mut hint = run.hint;
+                    let mut font_size = run.font_size;
+                    let mut transform = run.transform;
+                    if hint {
+                        // If hinting was requested and our transform matrix is just a uniform
+                        // scale, then adjust our font size and cancel out the matrix. Otherwise,
+                        // disable hinting entirely.
+                        if transform.matrix[0] == transform.matrix[3]
+                            && transform.matrix[1] == 0.0
+                            && transform.matrix[2] == 0.0
+                        {
+                            font_size *= transform.matrix[0];
```

Comment:
Can this also occur if the transform is a scale + rotation (i.e. `transform.matrix[1]==-transform.matrix[2]`?). Hinting (likely?) still needs to be disabled in that case.

Reply:
I believe it already rejects that case? Hinting should only be enabled if the diagonal components are equal and the others are zero. If the logic doesn’t capture that then it’s a bug.

We can potentially handle non-uniform scales as well if we’re only hinting in the Y direction (which is common), but I’ll leave that discussion for later.

Reply:
We've clarified this on Zulip; my understanding is that this translation would be valid; however it's equivalent to what we have now, so doesn't matter.